### ISSUE TITLE
[WD 0.3.4] ConcurrentModificationException on Server

### ISSUE THREAD

**Issue by No3371** [Jackiecrazy/WarDance#19]
This happened a while ago, I thought it was because of Performant (parallel processing entity) but as this happened again now without Performant installed, I guess there may be something to be resolved.

```
[14:42:13] [Server thread/ERROR] [ne.mi.ev.EventBus/EVENTBUS]: Exception caught during firing event: null
        Index: 8
        Listeners:
                0: HIGH
                1: ASM: class me.shedaniel.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                2: NORMAL
                3: ASM: class com.dynamic_view.event.EventHandler onDedicatedServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                4: net.minecraftforge.eventbus.EventBus$$Lambda$2699/1357058342@69b84c2d
                5: net.minecraftforge.eventbus.EventBus$$Lambda$2699/1357058342@26255209
                6: ASM: net.minecraftforge.common.ForgeInternalHandler@1bb9f74a onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                7: ASM: net.silentchaos512.scalinghealth.event.DamageScaling@208562d3 onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                8: ASM: class jackiecrazy.wardance.handlers.EntityHandler lure(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                9: ASM: class com.gerflargle.taffy.Description modifyDescription(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                10: ASM: com.github.alexthe666.citadel.server.CitadelEvents@64ed4a9 onServerTickEvent(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                11: ASM: class dev.itsmeow.whisperwoods.util.WWServerTaskQueue serverTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                12: ASM: class se.mickelus.tetra.ServerScheduler onPlayerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                13: ASM: class com.mrh0.buildersaddition.event.GameEvents tickEvent(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                14: ASM: me.ichun.mods.ichunutil.common.core.EventHandlerServer@36e7569c onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                15: ASM: class com.simibubi.create.events.CommonEvents onServerTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                16: ASM: net.infstudio.goki.common.handlers.TickHandler@5b40d7e0 serverTick(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
                17: ASM: com.sk89q.worldedit.forge.ThreadSafeCache@66951941 tickStart(Lnet/minecraftforge/event/TickEvent;)V
                18: ASM: me.lucko.spark.forge.ForgeTickHook@1be01b5d onTick(Lnet/minecraftforge/event/TickEvent;)V
                19: ASM: me.lucko.spark.forge.ForgeTickReporter@149e78f6 onTick(Lnet/minecraftforge/event/TickEvent;)V
                20: LOWEST
                21: ASM: class cofh.core.event.AreaEffectEvents handleTickEndEvent(Lnet/minecraftforge/event/TickEvent$ServerTickEvent;)V
java.util.ConcurrentModificationException
        at java.util.HashMap$HashIterator.remove(Unknown Source)
        at jackiecrazy.wardance.handlers.EntityHandler.lure(EntityHandler.java:177)
        at net.minecraftforge.eventbus.ASMEventHandler_284_EntityHandler_lure_ServerTickEvent.invoke(.dynamic)
        at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
        at net.minecraftforge.fml.hooks.BasicEventHooks.onPreServerTick(BasicEventHooks.java:115)
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:785)
        at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642)
        at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232)
        at java.lang.Thread.run(Unknown Source)

[14:42:13] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
java.util.ConcurrentModificationException: null
        at java.util.HashMap$HashIterator.remove(Unknown Source) ~[?:1.8.0_261] {}
        at jackiecrazy.wardance.handlers.EntityHandler.lure(EntityHandler.java:177) ~[wardance:0.3.4] {re:mixin,re:classloading}
        at net.minecraftforge.eventbus.ASMEventHandler_284_EntityHandler_lure_ServerTickEvent.invoke(.dynamic) ~[?:?] {}
        at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-4.0.0.jar:?] {}
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?] {}
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?] {}
        at net.minecraftforge.fml.hooks.BasicEventHooks.onPreServerTick(BasicEventHooks.java:115) ~[forge:?] {re:mixin,re:classloading}
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:785) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:enhancedcelestials.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
        at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:enhancedcelestials.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
        at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:enhancedcelestials.mixins.json:MixinMinecraftServer,pl:mixin:A,pl:runtimedistcleaner:A}
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_261] {}
```

**Comment by Jackiecrazy** [comment]
Fixed, thanks for reporting. The next version is still in the works, I trust this issue is not overly prevalent? If it is, I'll have to throw out a hotfix.

**Comment by No3371** [comment]
It does not happens frequently so I guess that's ok.

I'm actually glad about this issue comes within WarDance because I can test more on Performant... there seems other issues with Performant though, like mobs do not move toward players at all, only try to attack when player get very close.

**Comment by Jackiecrazy** [comment]
This should be fixed in 0.5.11.
